### DISCUSSION TITLE
Cannot get a base connection

### DISCUSSION THREAD

**Discussion by BennyDS** [Kegetys/simagic-arduino-poc#4] [Q&A]
Hi
I'm trying this out but its not working. I get the "nRF24 connected" message, but nothing in simpro manager or in DIView.
I'm thinking it might be firmware related?
I'm on Simpro Manager 2.0.338 and it says firmware 191 for the base. (Alpha mini)
Any ideas where to look?

Thx.

**Comment by Anguiano76** [comment]
Hello

In my case, changing in the code only the channel number (I use channel 80 which is the default in SimPro Manager) and using a Wemos D1 mini ESP8266 from aliexpress with an nRF24l01 also from aliexpress, it works for me and appears in the software as GT1/GT4. Two axles appear, clutch 1 and clutch 2 and in the state of the buttons another two, which fluctuate between several numbers, all with the B in front. 

I'll have to study the code, but I have no idea how to connect and configure the buttons.

Best regards

**Comment by BennyDS** [comment]
I never got my arduino to connect to the base. I kind of gave up on it and made my own wireless transmitter and receiver.
I might try it again using that Wemos board you used.
The part for the buttons is here. this is in the code to simulate button presses.

```
// invent some button presses
  const int buttons = 8;
  const bool state[buttons] = 
  {
    (millis() % 1000) < 500,
    (millis() % 5000) < 2500,
    (millis() % 7000) < 2000,
    (millis() % 100)  < 50,
    (millis() % 2600) < 1000,
    (millis() % 2600) < 1500,
    (millis() % 2600) < 2000,
    (millis() % 2600) < 2500,
  };
```

Replace these lines " (millis() % 1000) < 500" with something that reads your button state. In this example, 8 buttons. Add more lines if you have more buttons. So read your digital input either being 1 or 0. That's it :-)

**Reply by Anguiano76** [reply]
Thank you very much, I will try it. 

Out of curiosity, what are these lines? Test lines?

Thank you and best regards

**Comment by Anguiano76** [comment]
![IMG_20240921_180505835](https://github.com/user-attachments/assets/cac76673-5c70-4b65-902b-c27dd3059e32)

That's the only thing I checked for now.

**Comment by BennyDS** [comment]
Cool :-)

These lines basicly generate a 1 or a 0 at different time intervals.
this one:  "(millis() % 1000) < 500" is TRUE (so a 1 ==> button pressed) for the first half of every second, and FALSE (so 0 ==> button not pressed) for the second half of every second.

**Reply by Anguiano76** [reply]
Now I understand why the buttons pressed fluctuate. 

The truth is that I don't know much about programming, I can modify or see the logic of a code but I will have to look for how to add a line to monitor the status of a connected button. 

There are only a few free pins left on the Wemos. Could I add a button matrix there? 

Thanks for the answers, I was quite lost.

**Comment by BennyDS** [comment]
Honestly, ask chatgpt. :-) 
It does simple programming rather well. Also a good way to learn it. If you dont understand a line ask it to explain.

**Reply by Anguiano76** [reply]
I will do that, thanks for everything Benny

**Comment by Anguiano76** [comment]
BennyDS, if you get something, I would appreciate if you posted it here. I left the project on hold, but I will take it up again.Greetings and thanks

**Reply by BennyDS** [reply]
I did not look at it any further. It works with my own receiver so no point yet in trying to make it work. I might if I ever decide to get more steering wheels :-)
